src/script/mls/MLSMigration/MLSMigration.ts

```diff
@@ -137,40 +135,33 @@ const checkMigrationConfig = async (
 };
 
 interface MigrateConversationsToMLSParams {
-  apiClient: APIClient;
   core: Account;
   selfUserId: QualifiedId;
   conversationRepository: ConversationRepository;
-  isConversationOwnedBySelfTeam: (conversation: Conversation) => boolean;
+  userRepository: UserRepository;
+  teamState: TeamState;
 }
 
 const migrateConversationsToMLS = async ({
-  apiClient,
   core,
   selfUserId,
   conversationRepository,
-  isConversationOwnedBySelfTeam,
+  userRepository,
+  teamState,
 }: MigrateConversationsToMLSParams) => {
-  const conversations = conversationRepository.getLocalConversations();
+  //refetch all known users so we have the latest lists of the protocols they support
+  await userRepository.refreshAllKnownUsers();
 
   //TODO: implement logic for 1on1 conversations (both team owned and federated)
-  const regularGroupConversations = conversations.filter(
-    conversation =>
-      conversation.type() === CONVERSATION_TYPE.REGULAR &&
-      isConversationOwnedBySelfTeam(conversation) &&
-      !conversation.isTeam1to1(),
-  );
-
-  const {proteus: proteusConversations, mixed: mixedConversatons} =
-    groupConversationsByProtocol(regularGroupConversations);
```

Comment:
We cannot filter conversations by protocols here as the protocol might change during migration (proteus -> mixed -> mls). We're filtering conversations by protocol inside initialise, joinmixed and finalise methods instead.